pkg/kubelet/cm/devicemanager/topology_hints_test.go

```diff
@@ -677,11 +666,23 @@ type topologyHintTestCase struct {
 func getCommonTestCases() []topologyHintTestCase {
 	return []topologyHintTestCase{
 		{
-			description:   "Single Request, no alignment",
-			podUID:        "fakePod",
-			containerName: "fakeContainer",
-			request: map[string]string{
-				"testdevice": "1",
+			description: "Single Request, no alignment",
+			pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					UID: "fakePod",
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name: "fakeContainer",
+							Resources: v1.ResourceRequirements{
+								Limits: v1.ResourceList{
+									v1.ResourceName("testdevice"): resource.MustParse("1"),
+								},
+							},
+						},
+					},
+				},
```

Comment:
I'm not 100% opposed to this, but it does make each test case much more verbose.
What added flexibility does this give us over using the existing `makePod()` call and then overwriting what we need to as necessary?

Reply:
Well, test cases that are specific for `TestGetPodTopologyHints(...)` have a couple of containers inside them, that's why it is easier to define it as a raw pod (as `makePod()` returns a pod with one container).

I can try to create a similar function to `makePod()` for a multi-container pod, if needed.

See example these case's pod  below:
```
pod: &v1.Pod{
	ObjectMeta: metav1.ObjectMeta{
		UID: "fakePod",
	},
	Spec: v1.PodSpec{
		InitContainers: []v1.Container{
			{
				Resources: v1.ResourceRequirements{
					Limits: v1.ResourceList{
						v1.ResourceName("testdevice1"): resource.MustParse("1"),
						v1.ResourceName("testdevice2"): resource.MustParse("1"),
					},
				},
			},
			{
				Resources: v1.ResourceRequirements{
					Limits: v1.ResourceList{
						v1.ResourceName("testdevice1"): resource.MustParse("1"),
						v1.ResourceName("testdevice2"): resource.MustParse("2"),
					},
				},
			},
		},
		Containers: []v1.Container{
			{
				Name: "fakeContainer1",
				Resources: v1.ResourceRequirements{
					Limits: v1.ResourceList{
						v1.ResourceName("testdevice1"): resource.MustParse("1"),
						v1.ResourceName("testdevice2"): resource.MustParse("1"),
					},
				},
			},
			{
				Name: "fakeContainer2",
				Resources: v1.ResourceRequirements{
					Limits: v1.ResourceList{
						v1.ResourceName("testdevice1"): resource.MustParse("1"),
						v1.ResourceName("testdevice2"): resource.MustParse("1"),
					},
				},
			},
			{
				Name: "fakeContainer3",
				Resources: v1.ResourceRequirements{
					Limits: v1.ResourceList{
									v1.ResourceName("notRegistered"): resource.MustParse("1"),
								},
							},
						},
					},
				},
			},
```